docs/architecture/grpc-for-wcf-developers/rpc-types.md

```diff
@@ -0,0 +1,268 @@
+
+The following example shows a server streaming RPC.
+
+```protobuf
+service ClockStreamer {
+    rpc Subscribe(ClockSubscribeRequest) returns (stream ClockMessage);
+}
+```
+
+```csharp
+public class ClockStreamerService : ClockStreamer.ClockStreamerBase
+{
+    public override async Task Subscribe(ClockSubscribeRequest request,
+        IServerStreamWriter<ClockMessage> responseStream,
+        ServerCallContext context)
+    {
+        while (!context.CancellationToken.IsCancellationRequested)
+        {
+            var time = DateTimeOffset.UtcNow;
+            await responseStream.WriteAsync(new ClockMessage { message = $"The time is {time:t}." });
+            await Task.Delay(TimeSpan.FromSeconds(10), context.CancellationToken);
+        }
+    }
+}
+```
+
+This server stream could be consumed from a client application as shown in the following code:
+
+```csharp
+public async Task TellTheTimeAsync(CancellationToken token)
+{
+    var channel = GrpcChannel.ForAddress("https://localhost:5001");
+    var client = new ClockStreamer.ClockStreamerClient(channel);
+
+    var request = new ClockSubscribeRequest();
+    var response = client.Subscribe(request);
+
+    await foreach (var update in response.ResponseStream.ReadAllAsync(token))
+    {
+        Console.WriteLine(update.Message);
+    }
+}
+```
+
+> [!NOTE]
+> Server streaming RPCs are useful for subscription-style services, and also for sending very large datasets when it would be inefficient or impossible to build the entire dataset in memory. However, streaming responses isn't as fast as sending `repeated` fields in a single message, so as a rule streaming shouldn't be used for small datasets.
+
+### Differences to WCF
+
+A WCF duplex service uses a client callback interface that can have multiple methods. A gRPC server streaming service can only send messages over a single stream. If you need multiple methods, use a message type with either [an Any field or a oneof field](protobuf-any-oneof.md) to send different messages, and write code in the client to handle them.
+
+In WCF, the one-way method called from the client returns immediately, but the [ServiceContract](https://docs.microsoft.com/dotnet/api/system.servicemodel.servicecontractattribute?view=netframework-4.8) class with the session is kept alive until the connection is closed. In gRPC, the `Task` returned by the implementation method shouldn't complete until the connection is closed.
+
+## WCF one-way operations and gRPC client streaming
+
+WCF provides one-way operations (marked with `[OperationContract(IsOneWay = true)]`) that the client can call without waiting for a response. gRPC service methods always return a response, even if it's empty, and the client should always await that response. For "fire-and-forget" style messaging in gRPC, you can create a client streaming service.
```

Comment:
WCF waits for an ack (as mentioned in previous comment) which is sent just before dispatching. For WCF the difference between a void method and a one-way method is a void method returns an empty soap message once the operation is completed and a one-way method returns a transport specific ACK once the message has been verified as understood but before actual dispatch.